src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((acc, el) => {
+    const keyGroup = String(el[key]);
+
+    if (!acc[keyGroup]) {
+      acc[keyGroup] = [];
+    }
+
+    acc[keyGroup].push(el);
+
+    return acc;
+  }, {} as GroupsMap<T>);
```

Comment:
Using an empty object (`{}`) and casting it to `GroupsMap<T>` with `as` is a bit unsafe because it assumes the object will match the `GroupsMap<T>` type without actual validation. A safer way is to provide a more specific initial value that matches the `GroupsMap<T>` type. Consider initializing the accumulator with `Object.create(null)`, which creates a truly empty object without inherited properties, such as `toString` or `hasOwnProperty`, that might interfere with the grouping logic.